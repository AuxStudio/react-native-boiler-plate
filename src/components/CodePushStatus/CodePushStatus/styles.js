import { StyleSheet } from 'react-native';

import styleConstants from '../../../styleConstants';

const ICON_SIZE = 20;

const styles = StyleSheet.create({
  container: {
    alignSelf: 'stretch',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: styleConstants.colors.white,
    paddingVertical: styleConstants.dimensions.spacing.vertical / 2,
    paddingHorizontal: styleConstants.dimensions.spacing.horizontal / 2,
    borderRadius: styleConstants.dimensions.borderRadius,
  },
  text: {
    flex: 1,
    textAlign: 'center',
    ...styleConstants.fonts.types.small,
  },
  loader: {
    marginRight: styleConstants.dimensions.spacing.horizontal / 2,
  },
  icon: {
    fontSize: ICON_SIZE,
    color: styleConstants.colors.success,
    marginRight: styleConstants.dimensions.spacing.horizontal / 2,
  },
  warningIcon: {
    fontSize: ICON_SIZE,
    color: styleConstants.colors.danger,
    marginRight: styleConstants.dimensions.spacing.horizontal / 2,
  },
});

export default styles;
