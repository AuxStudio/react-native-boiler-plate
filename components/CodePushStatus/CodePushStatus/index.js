import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, ActivityIndicator } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

import styleConstants from '../../../styleConstants';
import styles from './styles';

const propTypes = {
  iconName: PropTypes.string,
  text: PropTypes.string,
  isLoading: PropTypes.bool,
};

const defaultProps = {};

const CodePushStatus = ({ iconName, text, isLoading }) => {
  const iconComponent = isLoading ? (
    <ActivityIndicator size="small" color={styleConstants.colors.green} style={styles.loader} />
  ) : (
    <Icon name={iconName} style={styles.icon} />
  );

  return (
    <View style={styles.container}>
      {iconComponent}

      <Text style={styles.text}>{text}</Text>
    </View>
  );
};

CodePushStatus.propTypes = propTypes;
CodePushStatus.defaultProps = defaultProps;

export default CodePushStatus;
