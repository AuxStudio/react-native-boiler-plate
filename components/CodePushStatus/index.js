import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import codePush from 'react-native-code-push';

import CodePushStatus from './CodePushStatus';

export class CodePushStatusContainer extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  static propTypes = {
    codePushStatus: PropTypes.number,
    codePushDownloadProgress: PropTypes.shape({
      receivedBytes: PropTypes.number,
      totalBytes: PropTypes.number,
    }),
  };

  static defaultProps = {};

  render() {
    const { codePushStatus, codePushDownloadProgress } = this.props;
    let iconName;
    let statusText;
    let isLoading;

    switch (codePushStatus) {
      case codePush.SyncStatus.UP_TO_DATE:
        statusText = 'The app is up to date';
        iconName = 'check';
        break;
      case codePush.SyncStatus.UPDATE_INSTALLED:
        statusText = 'Update ready. Close and reopen the app, to apply.';
        iconName = 'warning';
        break;
      case codePush.SyncStatus.CHECKING_FOR_UPDATE:
        statusText = 'Checking for app updates';
        isLoading = true;
        break;
      case codePush.SyncStatus.DOWNLOADING_PACKAGE:
        statusText = `Downloading update: ${
          codePushDownloadProgress
            ? Number(
                (codePushDownloadProgress.receivedBytes / codePushDownloadProgress.totalBytes) *
                  100,
              ).toFixed(0) // FIXME: Should be a util
            : 0
        }%`;
        isLoading = true;
        break;
      default:
        break;
    }

    return statusText ? (
      <CodePushStatus iconName={iconName} text={statusText} isLoading={isLoading} />
    ) : null;
  }
}

function mapStateToProps(state) {
  return {
    codePushStatus: state.appState.codePushStatus,
    codePushDownloadProgress: state.appState.codePushDownloadProgress,
  };
}

export default connect(mapStateToProps)(CodePushStatusContainer);
