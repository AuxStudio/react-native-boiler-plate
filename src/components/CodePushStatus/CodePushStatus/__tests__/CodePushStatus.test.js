import React from 'react';
import renderer from 'react-test-renderer';

import CodePushStatus from '..';

describe('CodePushStatus', () => {
  const iconName = 'check';
  const text = 'Up to date';

  describe('renders', () => {
    it('renders', () => {
      const component = renderer.create(<CodePushStatus iconName={iconName} text={text} />);

      expect(component).toMatchSnapshot();
    });

    it('renders the loading state', () => {
      const component = renderer.create(
        <CodePushStatus iconName={iconName} text={text} isLoading />,
      );

      expect(component).toMatchSnapshot();
    });
  });
});
