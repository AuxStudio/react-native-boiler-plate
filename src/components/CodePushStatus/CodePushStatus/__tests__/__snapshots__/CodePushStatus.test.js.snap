// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodePushStatus renders renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 10,
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingHorizontal": 7.5,
      "paddingVertical": 5,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "color": "#4caf50",
          "fontSize": 20,
          "marginRight": 7.5,
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    î—Š
  </Text>
  <Text
    style={
      Object {
        "color": "#212121",
        "flex": 1,
        "fontFamily": "System",
        "fontSize": 15,
        "textAlign": "center",
      }
    }
  >
    Up to date
  </Text>
</View>
`;

exports[`CodePushStatus renders renders the loading state 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 10,
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingHorizontal": 7.5,
      "paddingVertical": 5,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
    style={
      Object {
        "marginRight": 7.5,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#212121",
        "flex": 1,
        "fontFamily": "System",
        "fontSize": 15,
        "textAlign": "center",
      }
    }
  >
    Up to date
  </Text>
</View>
`;
